File: leetcode/java/7.ReverseInteger.java
class Main {
  static int reverse(int x){
   long n=x;
    long ans=0;
    int sign =1;
    if(x<0) {
      n*=-1;
      sign = -1;
      };
    while(n>0){
      ans= ans*10 + n%10;
      n/=10;
    }
    if(ans > Integer.MAX_VALUE){
        return 0;
    }     
    return (int)(sign*ans);  }

  public static void main(String[] args) {
    System.out.println(reverse(123));
  }
}

File: leetcode/kotlin/7.ReverseInteger.kt
class A{
    companion object{
        fun reverse(x:Int):Int{
            var n:Long = x.toLong();
            var ans:Long=0;
            var sign:Int=1;
            if(x<0){
                n*=-1;
                sign*=-1;
            }
            while(n>0){
                ans= ans*10 + n%10;
                n/=10;
            }
            if(ans > Integer.MAX_VALUE){
                return 0;
            }
            return (sign*ans).toInt();
        }
    }
}


fun main(){
    println(A.reverse(123))
}